otel/span_processor.go

```diff
@@ -103,6 +103,10 @@ func (ssp *sentrySpanProcessor) ForceFlush(ctx context.Context) error {
 
 func flushSpanProcessor(ctx context.Context) error {
```

Comment:
I guess one question is if anyone is actually going to pass a context in here that contain a hub?

Seems more like this will almost always be something like `context.Background()' with no hub. I think the original purpose here, at least looking at the otel project, is to allow the user to pass in a context with a timeout. The flush method though already has a timeout, so context passed here isn't really useful to us.

Reply:
You're right, it's not ideal, and relying on people passing the right context is not a good approach. 

It might be still good to let people pass sentry.Hub via context (for more flexibility, and maybe in the future allowing multiple span processor instances?), so your fix should be good for now 👍